contract/src/offerItems.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @file Contract to mint and sell a few Item NFTs at a time.
+ *
+ * We declare variables (including functions) before using them,
+ * so you may want to skip ahead and come back to some details.
+ * @see {start} for the main contract entrypoint
+ *
+ * @see {@link https://docs.agoric.com/guides/zoe/|Zoe Overview} for a walk-thru of this contract
+ * @see {@link https://docs.agoric.com/guides/js-programming/hardened-js.html|Hardened JavaScript}
+ * for background on `harden` and `assert`.
+ */
+// @ts-check
+
+import { Far } from '@endo/far';
+import { M, getCopyBagEntries, mustMatch } from '@endo/patterns';
+import { AmountMath, AssetKind } from '@agoric/ertp/src/amountMath.js';
+import { AmountShape } from '@agoric/ertp/src/typeGuards.js';
+import { atomicRearrange } from '@agoric/zoe/src/contractSupport/atomicTransfer.js';
+import '@agoric/zoe/exported.js';
+
+const { Fail, quote: q } = assert;
+
+// #region bag utilities
+/** @type { (xs: bigint[]) => bigint } */
+const sum = xs => xs.reduce((acc, x) => acc + x, 0n);
+
+/**
+ * @param {import('@endo/patterns').CopyBag} bag
+ * @returns {bigint[]}
+ */
+const bagCounts = bag => {
+  const entries = getCopyBagEntries(bag);
+  return entries.map(([_k, ct]) => ct);
+};
+// #endregion
+
+/**
+ * This contract is parameterized by terms for price and,
+ * optionally, a maximum number of items sold for that price (default: 3).
+ *
+ * @typedef {{
+ *   price: Amount;
+ *   maxItems?: bigint;
+ * }} ItemTerms
+ */
+
+/**
+ * Start a contract that
+ *   - creates a new non-fungible asset type for Items, and
+ *   - handles offers to buy up to `maxItems` items at a time.
+ *
+ * As is typical in Zoe contracts, the flow is:
+ *   1. contract does internal setup
+ *   2. client uses a public facet method -- {@link makeBuyItemsInvitation} in this case --
+ *      to make an invitation.
+ *   3. client makes an offer using the invitation, along with
+ *      a proposal (with give and want) and payments. Zoe escrows the payments, and then
+ *   4. Zoe invokes the offer handler specified in step 2 -- here {@link buyItemsHandler}.
+ *
+ * @param {ZCF<ItemTerms>} zcf
+ */
+export const start = async zcf => {
+  const { price, maxItems = 3n } = zcf.getTerms();
```

Comment:
In the 3n case, is the intent that `price` is enough for all three items, or were you expecting it to be `price * quantity`?

Reply:
price is enough for all 3